### PR TITLE
Add support for report transaction API

### PR DESCRIPTION
**Pull request by oalders**
I have tested this against a local copy of the API and there was no exception.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -222,6 +224,26 @@ WebServiceClient client = new WebServiceClient.Builder(6, "ABCD567890").build();
 System.out.println(client.insights(request));
 ```
 
+### Report Transaction
+
+```java
+import java.net.InetAddress;
+import java.util.UUID;
+import com.maxmind.minfraud.request.TransactionReport.Tag;
```
Comment (oschwald) [inline README.md:232]: I'd either make this example a complete class, remove the imports, or split them off. As is, this isn't valid Java code. I left off the imports for the other examples as people usually let their IDEs handle them, but I think there is a decent argument for making the examples directly runnable, if you want to go that route.

File: README.md [hunk 2]
```diff
@@ -222,6 +224,26 @@ WebServiceClient client = new WebServiceClient.Builder(6, "ABCD567890").build();
 System.out.println(client.insights(request));
 ```
 
+### Report Transaction
```
Comment (oschwald) [inline README.md:227]: It'd be nice to have some documentation too about what this is and maybe a link to the dev docs. Pat added some for the PHP API. I don't know if we need as much as he added, but just a short summary would be good.

File: README.md (hunk 1 above)
Comment (oalders) [reply README.md]: Thanks. I've removed the imports. If the IDEs will DWIM then that's good enough for me. :)

File: README.md (hunk 2 above)
Comment (oalders) [reply README.md]: I've stolen some of Pat's work.

**Pull request merged**